src/vm/database/clarity_db.rs

```diff
@@ -680,6 +716,41 @@ impl<'a> ClarityDatabase<'a> {
             .expect("Failed to get block data.")
     }
 
+    /// Gets the `StacksBlockId` for the *current* chain tip Stacks block height, converts this
+    /// `StacksBlockId` to a `SortitionId` and then gets the `BurnchainHeaderHash` from the
+    /// `BurnStateDB` using this `SortitionId`.
+    ///
+    /// Returns Some() if `0 <= height < current_burnblock_height`, and None otherwise.
+    pub fn get_burnchain_block_header_hash_for_burnchain_height(
+        &mut self,
+        burnchain_block_height: u32,
+    ) -> Option<BurnchainHeaderHash> {
```

Comment:
@kantai said "Basically, if the `supplied height is >= current_burn_block_height` it should return none."

I conceptually get why this is, I think, because we are "building the block" and the snapshot wouldn't exist in sqlite3 yet.

But, what is the literal API call for `current_burn_block_height`.